hack/pin-custom-bundle-dockerfile.sh

```diff
@@ -23,14 +24,25 @@ for MOD_PATH in $(go list -m -json all | jq -r '. | select(.Path | contains("ope
     REPO_URL="quay.io/openstack-k8s-operators"
     if [[ "$GITHUB_USER" != "openstack-k8s-operators" || "$BASE" == "$IMAGEBASE" ]]; then
         if [[ "$IMAGENAMESPACE" != "openstack-k8s-operators" || "${IMAGEREGISTRY}" != "quay.io" ]]; then
-            REPO_CURL_URL="https://${IMAGEREGISTRY}/api/v1/repository/${IMAGENAMESPACE}"
             REPO_URL="${IMAGEREGISTRY}/${IMAGENAMESPACE}"
+            # Quay registry v2 api does not return all the tags that's why keeping v1 for quay and v2
+            # for local registry
+            if [[ ${LOCAL_REGISTRY} -eq 1 ]]; then
+                REPO_CURL_URL="${IMAGEREGISTRY}/v2/${IMAGENAMESPACE}"
+            else
+                REPO_CURL_URL="https://${IMAGEREGISTRY}/api/v1/repository/${IMAGENAMESPACE}"
+            fi
         else
             REPO_CURL_URL="https://quay.io/api/v1/repository/${GITHUB_USER}"
             REPO_URL="quay.io/${GITHUB_USER}"
         fi
     fi
 
-    SHA=$(curl -s ${REPO_CURL_URL}/$BASE-operator-bundle/tag/ | jq -r .tags[].name | sort -u | grep $REF)
+    if [[ ${LOCAL_REGISTRY} -eq 1 && "$BASE" == "$IMAGEBASE" ]]; then
```

Comment:
I didn't get why you need `&& "$BASE" == "$IMAGEBASE" ` here, since this script runs against all modules. This would limit it to replace only IMAGEBASE.
We will also need to change IMAGEBASE to be a list, in order to test multiple operators build, before generating the custom bundle. But can be added in another patch.

Reply:
Btw, the idea behind IMAGEBASE var is to cover the scenario where bots propose PRs from OKO org. That scenario was not not working since GITHUB_USER is "openstack-k8s-operators" in those patches.

Reply:
@viroel Thank you for reviewing it. I have used ""$BASE" == "$IMAGEBASE"" to set get proper SHA value otherwise if I donot use. SHA logic fails if we set LOCAL_REGISTRY to 1 and use v2 api. It fails the bundle-build process.
I am not sure how can i avoid that. need help here.

Reply:
Yeah, that makes sense. Since all modules other modules will be on reg v1, and LOCAL_REGISTRY will be set 1 in all iterations.